lib/client/ippool_e2e_test.go

```diff
@@ -265,4 +261,52 @@ var _ = Describe("IPPool tests", func() {
 			},
 		),
 	)
+
+	Describe("Checking operations perform data validation", func() {
+		testutils.CleanEtcd()
+		c, _ := testutils.NewClient("")
+		var err error
+		valErrorType := reflect.TypeOf(cerrors.ErrorValidation{})
+
+		// Step-1: Test data validation occurs on create.
+		It("should invoke validation failure", func() {
+			By("Creating a pool with small CIDR")
+			_, err = c.IPPools().Create(&api.IPPool{
+				Metadata: api.IPPoolMetadata{CIDR: testutils.MustParseCIDR("10.10.10.0/30")},
+				Spec:     api.IPPoolSpec{},
+			})
+
+			Expect(err).To(HaveOccurred())
```

Comment:
Misleading though - since we reflect.TypeOf(nil) .... 

Better to be explicit.